src/federation.cpp

```diff
@@ -493,30 +641,39 @@ void ThreadBlockSigner(NodeContext& node)
             using std::swap;
             CBlock empty;
             swap(g_block_proposal, empty);
+            g_block_acks.clear();
             g_block_sigs.clear();
         }
 
-        if (!GenerateBlockProposal()) {
-            LogPrint(BCLog::FEDERATION, "%s: Unable to generate block proposal.  Terminating round early.\n", __func__);
+        int64_t sec_since_epoch = nextblocktime.time_since_epoch().count();
+        g_roundno = sec_since_epoch / 5;
```

Comment:
0107ee15747c01ea1de5c69498c42ee752a5bdb9

I am assuming several things here, but does this define the current round as the block time in seconds divided by 5 modulo member count? If so, won't this have a pretty poor distribution when there are e.g. 12 peers and 1 minute per block, since the peer will tend to be the same one each time? (12*5 = 60 s = 1 min = block interval).

Reply:
This is a mistake. We should be dividing by the length of the period (60 seconds) to get the current round, of course. During development I had this shortened to 5 seconds so I didn't have to sit around waiting for blocks to be generated. Looks like I accidentally committed that constant change at this point.

We should make the round parameters configurable, but that's a larger issue.

Reply:
Fixed the constant. Not marking as resolved until we decide how to make this configurable.

Reply:
Ahh, that makes sense. Can't you use `Params().GetConsensus().nPowTargetSpacing` directly?

Reply:
Well yes, that would be the obvious thing to do. Duh.